### ISSUE TITLE
Warning issued when running unit-tests on the GPU

### ISSUE THREAD

**Issue by frapac** [exanauts/ExaPF.jl#4]
I was testing `ExaPF` and get a few warnings issued when running the tests on the GPU. 
```julia
┌ Warning: calls to Base intrinsics might be GPU incompatible
│   exception =
│    You called atan(x::T) where T<:Union{Float32, Float64} in Base.Math at special/trig.jl:519, maybe you intended to call atan(x::Float64) in CUDA at /home/fpacaud/.julia/packag
es/CUDA/42B9G/src/device/cuda/math.jl:32 instead?
``` 

I guess you may be aware of this issue, as it arises when running the test. It looks like the issue is related to `GPUArrays`.

My system is:
```
julia> versioninfo()
Julia Version 1.4.1
Commit 381693d3df* (2020-04-14 17:20 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
```

**Comment by michel2323** [comment]
This is related to #7 . The code was first generated using GPUs. When switching from target = "cpu" to target = "cuda" only parts of the code will run on the GPU as some code has been generated for CPU.

**Comment by frapac** [comment]
Solved by #15
